### PR TITLE
CSS Usage Cleanup

### LINKED ISSUE #6530: CSS Usage Cleanup
Follow up to #6395 

I'll flesh this out more tomorrow:

- [x] add style fields to phosphor packages with styles
- template index.css ( in examples too)
  -- need to declare which other deps to import (like font-awesome)
  -- then we add them in dependency order
  -- then we add ./base.css
- Specify where css file was generated from

### PR DESCRIPTION
**Pull request by blink1073**
## References
Fixes #6530 

## Code changes
Templates `style/index.css` based on dependencies that have `style` fields, in dependency order
Adds `extraStyles` metadata in `package.json` for additional external style imports.
Ignore the temporary `imports.css` file when building core mode.

## User-facing changes
Frees the developer from maintaining the style index in most cases, with some manual 
intervention for additional imports and imports to skip.
Adds developer docs for the new process.
 

## Backwards-incompatible changes
None

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -1618,14 +1618,22 @@
   resolved "https://registry.yarnpkg.com/@phosphor/domutils/-/domutils-1.1.3.tgz#5aeeaefb4bbfcc7c0942e5287a29d3c7f2b1a2bc"
   integrity sha512-5CtLAhURQXXHhNXfQydDk/luG1cDVnhlu/qw7gz8/9pht0KXIAmNg/M0LKxx2oJ9+YMNCLVWxAnHAU0yrDpWSA==
 
-"@phosphor/dragdrop@^1.3.0", "@phosphor/dragdrop@^1.3.1":
+"@phosphor/dragdrop@^1.3.1":
```
Comment (jasongrout) [inline yarn.lock]: Looks like we need to dedupe to get rid of one of this one.

File: yarn.lock [hunk 2]
```diff
@@ -1675,6 +1683,23 @@
     "@phosphor/signaling" "^1.2.3"
     "@phosphor/virtualdom" "^1.1.3"
 
+"@phosphor/widgets@^1.7.1":
```
Comment (jasongrout) [inline yarn.lock]: Another dedupe needed (cf 1.7.0 above).

**Comment by blink1073** [comment]
The logic is almost done. I am going to defer the examples for after the RC. 
Note to self:
We can't reuse the graph for build because it relies on local paths
We should add private data to graph to avoid reading the package.json files twice

**Comment by jasongrout** [comment]
@blink1073 - Ready for review?

Looks like the doc failure is unrelated - a link to binderhub down.

File: buildutils/src/ensure-repo.ts [hunk 3]
```diff
@@ -250,6 +249,42 @@ export async function ensureIntegrity(): Promise<boolean> {
...
   });
 
+  // Build up an ordered list of CSS imports for each local package.
+  Object.keys(locals).forEach(name => {
+    const data = pkgData[name];
+    const deps: Dict<string> = data.dependencies || {};
+    const skip = SKIP_CSS[name] || [];
+    const cssData: Dict<Array<string>> = {};
+
+    if (data.jupyter && data.jupyter.extraStyles) {
```
Comment (vidartf) [inline buildutils/src/ensure-repo.ts]: Should this not be `data.jupyterlab`? I see there are some extra styles missing somewhere (codemirror / xterm), while others are included (blueprint).

Comment (jasongrout) [reply buildutils/src/ensure-repo.ts]: I think you're right.

Comment (jasongrout) [reply buildutils/src/ensure-repo.ts]: Fixed in #6541

**Pull request merged**